api/models/datastore.go

```diff
@@ -10,10 +10,9 @@ type Datastore interface {
 	// GetApp gets an App by name.
 	// Returns ErrDatastoreEmptyAppName for empty appName.
 	// Returns ErrAppsNotFound if no app is found.
-	GetApp(ctx context.Context, app *App) (*App, error)
-
+	GetAppByName(ctx context.Context, appName string) (*App, error)
 	GetAppByID(ctx context.Context, appID string) (*App, error)
-
+	EnsureApp(ctx context.Context, appName string) (string, error)
```

Comment:
comment please describing the contract here, with proper GoDoc format `EnsureApp will...`. fwiw `GetAppID(ctx, appName) (id string, err error)` is more clear to me than `EnsureApp` which conveniently returns an ID and is more uniform with the rest of the methods on this interface ($.02)

Reply:
i guess this is an overlap of `GetAppByName` already kinda?

Reply:
There's no overlap, take a look at sql implementation

Reply:
from what I can tell, GetAppByID and GetAppID are mostly the same with a slightly different signature. we can clean up later, that works for now.

Reply:
So, `GetAppByID` returns app full object (there might be another question whether it is necessary to get the whole app object instead of just ID string), including name config when `GetAppID` returns ID only.

Reply:
it's true, we don't have to get the config. it's still not clear to me exactly what remains broken in the front end, probably either way would work but I'm not sure with our 2 layers of middleware to appease. `getAppTx` itself seems extremely hairy

Reply:
Why `getAppTx` looks hairy? It just accepts the query and tried to run that with unmarshaling to an App structure? Where do you see it as bad way?

Reply:
if there is any bug and for whatever reason, say it always looks up by name, then it will be a security flaw. it runs multiple different queries through the same function, and the sqlx interface is already pretty minimal, there's just no need for it. we could have a helper for every query, but it doesn't really give us any sugar over `db.Exec`